includes/openid-connect-generic-client.php

```diff
@@ -407,6 +407,12 @@ function validate_user_claim( $user_claim, $id_token_claim ) {
 
 		// allow for other plugins to alter the login success
 		$login_user = apply_filters( 'openid-connect-generic-user-login-test', true, $user_claim );
+
+        // make sure they should have access
+		$roles = $user_claim['roles'];
+		if (!in_array('developer', $roles) and !in_array('partner', $roles) and !in_array('sales_person', $roles)) {
```

Comment:
I wouldn't put this here tbh, this has nothing to do with oauth2/openid connect. I would add it as a check in specific to your app. mainly it will cause issues if this is used in another wordpress project